### PR TITLE
fio: convert to json for metric data

### PR DESCRIPTION
**Pull request by atheurer**
Use same libaries uperf uses to process data

### CODE REVIEW

**Comment by portante** [comment]
[Create JSON result files from pbench fio](https://trello.com/c/etAG2yza/8-create-json-result-files-from-pbench-fio)

File: agent/bench-scripts/pbench-fio [hunk 1]
```diff
@@ -641,8 +604,6 @@ function fio_run_benchmark() {
 			for block_size in `echo $block_sizes | sed -e s/,/" "/g`; do
 				job_num=1
 				iteration="${count}-${test_type}-${block_size}KiB"
-				   
-                                echo $iteration >> $benchmark_iterations
```
Comment (portante) [inline agent/bench-scripts/pbench-fio]: @ndokos, don't we need this now?  Or is this being replaced by another method?

File: agent/bench-scripts/pbench-fio [hunk 2]
```diff
@@ -372,17 +372,14 @@ function fio_process_options() {
...
-			err_log "I need a directory if postprocessing an existing result (--run-dir=)"
-			exit 1
+			echo "I need a directory if postprocessing an existing result (--run-dir=)"
+		else
+			benchmark_run_dir="$run_dir"
 		fi
-		benchmark_fullname="$(basename $run_dir)"
-		benchmark_run_dir="$run_dir"
 	fi
-	benchmark_iterations="$pbench_tmp/${benchmark_fullname}.iterations"
```
Comment (portante) [inline agent/bench-scripts/pbench-fio]: I think we need this for the indexing work.  Is there a replacement for this?

File: agent/bench-scripts/postprocess/fio-postprocess [hunk 3]
```diff
@@ -1,201 +1,292 @@
...
+			my $tool_dir_id = $this_tool_dir;
+			$tool_dir_id =~ s/^$tool_label_pattern//;
+			$this_tool_dir = $tool_group_dir . "/" . $this_tool_dir;
+			$tool_ids{$this_tool_dir} = $tool_dir_id;
 		}
 	}
+} else {
+	print "$script: could not find any directories in $tool_group_dir which matched $tool_label_pattern\n";
+}
+closedir  $dh;
```
Comment (ndokos) [inline agent/bench-scripts/postprocess/fio-postprocess]: Should the closedir() be in the if branch? I get complaints about invalid dir handles.

File: agent/bench-scripts/pbench-fio (hunk 1 above)
Comment (ndokos) [inline agent/bench-scripts/pbench-fio]: Yes, but the record_iteration function is not in master yet (only in wip-indexing). I'l fix this up after I PR the iterations commit.

**Comment by ndokos** [comment]
We'll need to add the Switch.PM module as a dependency to the RPM. We'll also need to add it the Travis environment for unit testing.

File: agent/bench-scripts/postprocess/fio-postprocess (hunk 3 above)
Comment (atheurer) [inline agent/bench-scripts/postprocess/fio-postprocess]: You are correct, it should be moved within the branch

File: agent/bench-scripts/pbench-fio (hunk 2 above)
Comment (atheurer) [inline agent/bench-scripts/pbench-fio]: I am not sure, but adding it back should not be a problem for this PR

**Comment by portante** [comment]
@atheurer, @ndokos, where are we at with this work?

**Comment by ndokos** [comment]
Superseded by PR #590.

**Pull request closed without merging**